apps/src/templates/instructions/TeacherFeedback.jsx

```diff
@@ -225,173 +134,85 @@ export class TeacherFeedback extends Component {
-    let timeStyle;
-    let time;
-    let feedbackMessage;
-    let currentTeacherStudentSeen = false;
-
-    //Student view indicates when feedback was last updated
-    if (this.props.viewAs === ViewType.Student) {
-      timeStyle = styles.timeStudent;
-      time = moment.min(moment(), moment(latestFeedback.created_at)).fromNow();
-      feedbackMessage = i18n.lastUpdated();
-    } else if (this.props.viewAs === ViewType.Teacher) {
-      timeStyle = styles.timeTeacher;
-      //Teacher view if current teacher left feedback & student viewed
-      if (studentSawFeedback) {
-        timeStyle = {
-          ...timeStyle,
-          ...styles.timeTeacherStudentSeen
-        };
-        feedbackMessage = i18n.seenByStudent();
-        time = this.getFriendlyDate(latestFeedback.student_seen_feedback);
-        currentTeacherStudentSeen = true;
-      } else {
-        //Teacher view if current teacher left feedback & student did not view
-        feedbackMessage = i18n.lastUpdatedCurrentTeacher();
-        time = this.getFriendlyDate(latestFeedback.created_at);
-      }
-    }
-    return this.displayFeedbackMessage([
-      timeStyle,
-      feedbackMessage,
-      currentTeacherStudentSeen,
-      time
-    ]);
-  };
-
   render() {
-    const latestFeedback = this.latestFeedback();
+    const {
+      verifiedTeacher,
+      viewAs,
+      rubric,
+      visible,
+      displayKeyConcept,
+      disabledMode
+    } = this.props;
+
+    const {
+      comment,
+      performance,
+      latestFeedback,
+      submitting,
+      errorState
+    } = this.state;
+
     const feedbackUnchanged = this.feedbackIsUnchanged();
 
     const buttonDisabled =
       feedbackUnchanged ||
-      this.state.submitting ||
-      this.state.errorState === ErrorType.Load ||
-      !this.props.verifiedTeacher;
+      submitting ||
+      errorState === ErrorType.Load ||
+      !verifiedTeacher;
+
     const buttonText = latestFeedback ? i18n.update() : i18n.saveAndShare();
 
-    const placeholderWarning = this.props.verifiedTeacher
+    const placeholderWarning = verifiedTeacher
       ? i18n.feedbackPlaceholder()
       : i18n.feedbackPlaceholderNonVerified();
 
     const placeholderText =
       latestFeedback && latestFeedback.comment
         ? latestFeedback.comment
         : placeholderWarning;
-    const dontShowStudentComment =
-      !this.state.comment && this.props.viewAs === ViewType.Student;
-
-    const showFeedbackInputAreas =
-      !this.props.displayKeyConcept &&
-      !(!this.state.performance && this.props.viewAs === ViewType.Student);
 
-    const rubricLevels = [
-      'performanceLevel1',
-      'performanceLevel2',
-      'performanceLevel3',
-      'performanceLevel4'
-    ];
-
-    const studentSawFeedback =
-      this.state.latestFeedback.length > 0 &&
-      !!latestFeedback.student_seen_feedback;
+    const dontShowStudentComment = !comment && viewAs === ViewType.Student;
```

Comment:
Maybe do one of the following?
1) Change this variable to showStudentComment and incorporate the check for displayKeyConcept, or
2) Move the whole check inline and get rid of the variable